phabricatoremails/query_position_store.py

```diff
@@ -0,0 +1,16 @@
+from dataclasses import dataclass
+
+from phabricatoremails.models import QueryPosition
+from sqlalchemy.orm import Session
+
+
+@dataclass
+class QueryPositionStore:
+    _db_session: Session
+
+    def get_position(self):
+        return self._db_session.query(QueryPosition).one()
+
+    def set_initial_position(self, position: int):
```

Comment:
This would be more user friendly if it checked with `get_position` and raised an exception if the value was already set, rather than relying on the database to raise an exception.

Reply:
I think that relying on the database for constraint-checking is an ok pattern.

For example, if you are working on a service for user account signup, and email addresses have to be unique, you probably wouldn't query out all the email addresses and verify this constraint in your programming language, you'd let your database do the heavy lifting.

Reply:
> Even Github's signup page works this way, and it is doubtful that they're running insert statements on every single keypress, given that they are more expensive than select statements.

This is a little bit different - of course GitHub would use something like `SELECT` there because they don't want to `INSERT` every subset of your nickname that you're entering.

>  it's ambiguous whether this is designed to be used once and only once to set an "initial" position that is used by your application, or to be reused over and over to reset the "initial" position of something else

I'll add a docstring :+1: